src/util.ts

```diff
@@ -769,19 +770,81 @@ export function validateMinimumIncrease(proposed: string, min: string) {
 }
 
 /**
- * Extracts content identifier from ipfs url.
+ * Removes ipfs protocol prefix from ipfs-url.
  *
- * @param url - Ipfs url.
- * @returns Ipfs content identifier as string.
+ * @param ipfsUrl - ipfs url
+ * @returns Ipfs content identifier and (possibly) path in a string
+ * @throws will throw if the url passed is not ipfs.
  */
-export function getIpfsUrlContentIdentifier(url: string): string {
-  if (url.startsWith('ipfs://ipfs/')) {
-    return url.replace('ipfs://ipfs/', '');
-  }
+export function removeIpfsProtocolPrefix(ipfsUrl: string) {
+  if (ipfsUrl.startsWith('ipfs://ipfs/')) {
+    return ipfsUrl.replace('ipfs://ipfs/', '');
+  } else if (ipfsUrl.startsWith('ipfs://')) {
+    return ipfsUrl.replace('ipfs://', '');
+  }
+  // this method should not be used with non-ipfs urls (i.e. startsWith('ipfs://') === true)
+  throw new Error('this method should not be used with non ipfs urls');
+}
 
-  if (url.startsWith('ipfs://')) {
-    return url.replace('ipfs://', '');
+/**
+ * Extracts content identifier and path from ipfs url.
+ *
+ * @param ipfsUrl - ipfs url
+ * @returns Ipfs content identifier as string and path as string.
+ * @throws Will throw if the url passed is not ipfs.
+ */
+export function getIpfsCIDv1AndPath(
+  ipfsUrl: string,
+): { cid: string; path?: string } {
+  const url = removeIpfsProtocolPrefix(ipfsUrl);
+
+  // check if there is a path
+  // (CID is everything preceding first forward slash, path is everything after)
+  const index = url.indexOf('/');
+  const cid = index !== -1 ? url.substring(0, index) : url;
+  const path = index !== -1 ? url.substring(index) : '';
+
+  // we want to ensure that the CID is v1 (https://docs.ipfs.io/concepts/content-addressing/#identifier-formats)
```

Comment:
Nit: Maybe worth mentioning why this is important to us specifically: because v0 isn't compatible with subdomain IPFS gateway URLs, but v1 is.

Reply:
Agreed, but as a side note, apparently this isn't always the case. Found some instances where v0 CIDS worked in the subdomain format. Will add to this note anyways!

Reply:
added to his comment here: https://github.com/MetaMask/controllers/pull/655/commits/3432359baba873cea2947164581603277438f990

Reply:
Really? Since v0 is mixed-case, it seemed unlikely that any would be completely lower case. But I guess that's possible.